### PR TITLE
Kernel migration 5.2.0-m4

### PR DESCRIPTION
**Pull request by Yasara123**

### CODE REVIEW

File: components/org.wso2.carbon.metrics.core/src/main/java/org/wso2/carbon/metrics/core/internal/MetricsComponent.java [hunk 1]
```diff
@@ -147,7 +154,11 @@ protected void setMetricsExtension(MetricsExtension metricsExtension) {
         if (logger.isDebugEnabled()) {
             logger.debug("Activating Metrics Extension {}", metricsExtension.getClass().getName());
         }
-        metricsExtension.activate(configProvider, metrics.getMetricService(), metrics.getMetricManagementService());
+        if (configProvider == null || metrics == null) {
```
Comment (chrishantha) [inline components/org.wso2.carbon.metrics.core/src/main/java/org/wso2/carbon/metrics/core/internal/MetricsComponent.java:157]: Why did you add this if condition??

File: distribution/deployment.yaml [hunk 2]
```diff
@@ -1,28 +1,29 @@
-# Copyright 2017 WSO2 Inc. (http://wso2.org)
+################################################################################
+#   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved
```
Comment (chrishantha) [inline distribution/deployment.yaml:2]: There is no need to change the copyright year and the license. It was already there. Please remove your changes

File: distribution/pom.xml [hunk 3]
```diff
@@ -165,7 +201,7 @@
                             <goal>publish-product</goal>
                         </goals>
                         <configuration>
-                            <productConfigurationFile>${basedir}/carbon.product</productConfigurationFile>
+                            <!--<productConfigurationFile>${basedir}/carbon.product</productConfigurationFile>-->
```
Comment (chrishantha) [inline distribution/pom.xml:204]: Why did you comment this?

File: tests/osgi-tests/src/test/java/org/wso2/carbon/metrics/osgi/MetricsTest.java [hunk 4]
```diff
@@ -78,77 +87,90 @@ private Bundle getBundle(String name) {
...
+        };
+    }
     @Test
     public void testMetricsCoreBundle() {
         Bundle coreBundle = getBundle("org.wso2.carbon.metrics.core");
         Assert.assertEquals(coreBundle.getState(), Bundle.ACTIVE);
     }
 
-    @Test
+    @Test(dependsOnMethods = { "testMetricsCoreBundle" })
```
Comment (chrishantha) [inline tests/osgi-tests/src/test/java/org/wso2/carbon/metrics/osgi/MetricsTest.java:109]: What's the reason for adding "dependsOnMethods"? We usually should avoid depending on other methods as much as possible.

File: tests/test-distribution/carbon-home/conf/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,259 @@
+################################################################################
+#   Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved
```
Comment (chrishantha) [inline tests/test-distribution/carbon-home/conf/deployment.yaml:2]: Copyright year should be 2017.

File: tests/test-distribution/pom.xml [hunk 6]
```diff
@@ -38,30 +38,110 @@
...
+            <type>zip</type>
+        </dependency>
     </dependencies>
 
     <build>
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-dependency-plugin</artifactId>
-                <version>${maven.dependency.plugin.version}</version>
```
Comment (chrishantha) [inline tests/test-distribution/pom.xml:48]: Why did you remove the version?

File: tests/test-distribution/pom.xml [hunk 7]
```diff
@@ -38,30 +38,110 @@
...
                             <goal>unpack</goal>
                         </goals>
                         <configuration>
                             <artifactItems>
                                 <artifactItem>
-                                    <groupId>org.wso2.carbon.metrics</groupId>
-                                    <artifactId>wso2carbon-metrics</artifactId>
+                                    <groupId>org.eclipse.equinox</groupId>
+                                    <artifactId>org.eclipse.equinox.executable</artifactId>
+                                    <version>3.5.0.v20110530-7P7NFUFFLWUl76mart</version>
```
Comment (chrishantha) [inline tests/test-distribution/pom.xml:141]: Version is hard coded. Is it possible to use a property?

File: tests/test-distribution/pom.xml [hunk 8]
```diff
@@ -38,30 +38,110 @@
...
                                     <type>zip</type>
                                     <overWrite>true</overWrite>
-                                    <outputDirectory>${project.build.directory}</outputDirectory>
+                                    <outputDirectory>target</outputDirectory>
```
Comment (chrishantha) [inline tests/test-distribution/pom.xml:144]: ${project.build.directory} is the target. This change is not needed.

File: tests/test-distribution/pom.xml [hunk 9]
```diff
@@ -182,4 +396,4 @@
             </plugin>
         </plugins>
     </build>
-</project>
+</project>
```
Comment (chrishantha) [inline tests/test-distribution/pom.xml:399]: Keep the new line at the end of file.

File: components/org.wso2.carbon.metrics.core/src/main/java/org/wso2/carbon/metrics/core/Gauge.java [hunk 10]
```diff
@@ -16,7 +16,7 @@
 package org.wso2.carbon.metrics.core;
 
 /**
- * A metric to read a particular value
+ * A metric to read a particular value.
```
Comment (chrishantha) [inline components/org.wso2.carbon.metrics.core/src/main/java/org/wso2/carbon/metrics/core/Gauge.java:19]: Did you do this on purpose? Adding a full stop for every Java Doc comment? This is okay, but I just want to know. Many classes have changed just because of adding a full stop to Java Doc comments.

Comment (Yasara123) [reply components/org.wso2.carbon.metrics.core/src/main/java/org/wso2/carbon/metrics/core/Gauge.java:19]: Because of check style errors.

**Review by chrishantha — APPROVED** [review]
I talked to @Yasara123 and got feedback for all changes. The changes are okay.
